### DISCUSSION TITLE
Some HELP please - Call to undefined method App\Models\User::isEmailVerificationRequired()

### DISCUSSION THREAD

**Discussion by bercksonjohnslyjeanlouis** [filamentphp/filament#10644] [General]
My filament app doesn't accepting any command and doesn't lauch. With this error...

`php artisan serve

   BadMethodCallException

  Call to undefined method App\Models\User::isEmailVerificationRequired()

  at vendor\laravel\framework\src\Illuminate\Support\Traits\ForwardsCalls.php:67
     63▕      * @throws \BadMethodCallException
     64▕      */
     65▕     protected static function throwBadMethodCallException($method)
     66▕     {
  ➜  67▕         throw new BadMethodCallException(sprintf(
     68▕             'Call to undefined method %s::%s()', static::class, $method
     69▕         ));
     70▕     }
     71▕ }

  i   Bad Method Call: Did you mean App\Models\User::getEmailForVerification() ?

  1   vendor\laravel\framework\src\Illuminate\Support\Traits\ForwardsCalls.php:36
      Illuminate\Database\Eloquent\Model::throwBadMethodCallException("isEmailVerificationRequired")

  2   vendor\laravel\framework\src\Illuminate\Database\Eloquent\Model.php:2334
      Illuminate\Database\Eloquent\Model::forwardCallTo(Object(Illuminate\Database\Eloquent\Builder), "isEmailVerificationRequired")`

**Comment by wq9578** [comment]
Trace back the stack to the point where isEmailVerificationRequired() is called is called, any why this is so.

You must have something like $user->isEmailVerificationRequired() in your code.

There are only two places in the current Filament code where isEmailVerificationRequired() is called, and they only would be relevant if your class User is derived from class Panel:

```
namespace Filament\Resources;
...
abstract class Resource
...
    public static function isEmailVerificationRequired(Panel $panel): bool
    {
        return $panel->isEmailVerificationRequired();
    }
```

```
namespace Filament\Pages\Concerns;
...
trait HasRoutes
...
    public static function isEmailVerificationRequired(Panel $panel): bool
    {
        return $panel->isEmailVerificationRequired();
    }
```

**Reply by bercksonjohnslyjeanlouis** [reply]
Where  must have something like $user->isEmailVerificationRequired() in your code ?

Because the two places mentioned are correct on my code...

**Reply by bercksonjohnslyjeanlouis** [reply]
But now i get this error  

` Non-static method App\Models\User::isEmailVerificationRequired() cannot be called statically

  at vendor\filament\filament\src\Resources\Pages\Page.php:91
     87▕     }
     88▕
     89▕     public static function isEmailVerificationRequired(Panel $panel): bool
     90▕     {
  ➜  91▕         return static::getResource()::isEmailVerificationRequired($panel);
     92▕     }
     93▕
     94▕     public static function getTenantSubscribedMiddleware(Panel $panel): string
     95▕     {

  1   vendor\filament\filament\src\Pages\Concerns\HasRoutes.php:46
      Filament\Resources\Pages\Page::isEmailVerificationRequired(Object(Filament\Panel))

  2   vendor\filament\filament\src\Resources\Pages\Page.php:79
      Filament\Pages\Page::getRouteMiddleware(Object(Filament\Panel))

PS C:\DEV\FILAMENT>
`

**Comment by l-mtc** [comment]
@bercksonjohnslyjeanlouis did you get this resolved?

**Comment by Capevace** [comment]
For anyone coming past this while googling for the error:

In my resource pages (`ListModelABC`, `EditModelABC`, etc.) I referenced the model in the `static $resource` instead of the resource. Fixed it by just using the correct class

**Comment by GreccoOliva-Franco** [comment]
Each Page of DemoResource, must have the line:

```php
// example for ViewRecord

class DemoPage extends ViewRecord {
  protected static string $resource = DemoResource::class;

// ...
}
```
